src/socket_poller.hpp

```diff
@@ -53,15 +53,19 @@ namespace zmq
             int fd;
 #endif
             void *user_data; 
+            short events;
         } event_t;
 
-        int add_socket (void *socket, void *user_data);
+        int add_socket (void *socket, void *user_data, short events);
+        int modify_socket (void *socket, short events);
         int remove_socket (void *socket);
 #if defined _WIN32
-        int add_fd (SOCKET fd, void *user_data);
+        int add_fd (SOCKET fd, void *user_data, short events);
+        int mofify_fd (SOCKET fd, short events);
```

Comment:
thanks. Probably explaining why it is failing on windows.
Fixed in a new commit.